creator-node/src/routes/tracks.js

```diff
@@ -37,12 +40,90 @@ const RehydrateIpfsQueue = require('../RehydrateIpfsQueue')
 const { FileProcessingQueue } = require('../FileProcessingQueue')
 const DBManager = require('../dbManager')
 const { generateListenTimestampAndSignature } = require('../apiSigning.js')
+const DiskManager = require('../diskManager')
 
+const server = new tus.Server()
 const readFile = promisify(fs.readFile)
+const tmpTrackArtifactsPath = DiskManager.getTmpTrackUploadArtifactsPath()
+const resumableUploadRoute = path.join(tmpTrackArtifactsPath, '*', '*')
+
+// Structure: <uuid>.<file_extension>. e.g. 1234-1234-1234-1234.mp3
+const getFileName = (req) => {
+  return req.headers.randomfilename + getFileExtension(req.headers.filename)
+}
 
 const SaveFileToIPFSConcurrencyLimit = 10
 
 module.exports = function (app) {
+  app.post('/track_content_upload', authMiddleware, ensurePrimaryMiddleware, ensureStorageMiddleware, syncLockMiddleware, checkFileMiddleware, async function (req, res, next) {
+    // Use the tus-node-server package to handle track upload
+
+    // Save file under randomly named folders to avoid collisions
+    const fileDir = path.join(tmpTrackArtifactsPath, req.headers.randomfilename)
+
+    try {
+    // Create directories for original file and segments
+      fs.mkdirSync(fileDir, { recursive: true })
+      fs.mkdirSync(fileDir + '/segments')
+
+      req.logger.info(`Created track disk storage: ${fileDir}, ${req.headers.randomfilename}`)
+
+      server.datastore = new tus.FileStore({
+        path: fileDir,
+        // Custom file naming
+        namingFunction: getFileName
+      })
+
+      // Initialze resumable upload flow
+      const resp = await server.handle.bind(server)(req, res, next)
+
+      if (resp.statusCode > 299 || resp.statusCode < 200) {
+        // TODO: add resp details
+        throw new Error(`Unsuccessful upload creation. fileDir=${fileDir} fileName=${req.headers.randomfilename}`)
+      }
+    } catch (e) {
+      return sendResponse(req, res, errorResponseServerError(e.toString()))
+    }
+  })
+
+  // Route that handles the resumable upload HEAD and PATCH requests.
+    try {
```

Comment:
spacing. try is indented too far. but also i'd like to dig into this a bit more tomorrow to understand how this works